core/lib/patternlab.js

```diff
@@ -343,111 +345,119 @@ var patternlab_engine = function (config) {
       cacheBuster: patternlab.cacheBuster
     });
 
+    // FIXME This simply iterates over all patterns, now we need to iterate over changed files
+    // with a defined order
+
+
+    var patternsToBuild = patternlab.patterns;
+
     //render all patterns last, so lineageR works
-    patternlab.patterns.forEach(function (pattern) {
+    patternsToBuild.forEach( pattern => renderSinglePattern(pattern, head));
```

Comment:
I factored out this method, because the iteration order will be made deterministic later on when walking the graph.